modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/components/sidebar/sections/actions/ActionsInfo.js

```diff
@@ -0,0 +1,342 @@
+
+	const [description, setDescription] = useState('');
+	const [executionType, setExecutionType] = useState('');
+	const [name, setName] = useState('');
+	const [priority, setPriority] = useState(DEFAULT_LIMIT);
+	const [template, setTemplate] = useState('');
+
+	useEffect(() => {
+		if (selectedItem.type === 'task' && executionTypeOptions.length < 3) {
+			executionTypeOptions.push({
+				label: Liferay.Language.get('on-assignment'),
+				value: 'onAssignment',
+			});
+		}
+
+		sortElements(executionTypeOptions, 'value');
+
+		return function cleanup() {
+			executionTypeOptions = executionTypeOptions.filter(({value}) => {
+				return value !== 'onAssignment';
+			});
+		};
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, []);
+
+	useEffect(() => {
+		const currentSectionData =
+			selectedItem.data.actions?.sectionsData[index];
+
+		setDescription((prev) =>
+			currentSectionData
+				? currentSectionData?.description
+				: prev.description ?? ''
+		);
+		setExecutionType((prev) =>
+			currentSectionData
+				? currentSectionData?.executionType
+				: prev.executionType ?? executionTypeOptions[0].value
+		);
+		setName((prev) =>
+			currentSectionData ? currentSectionData?.name : prev.name ?? ''
+		);
+		setPriority((prev) =>
+			currentSectionData
+				? currentSectionData?.priority
+				: prev.priority ?? ''
+		);
+		setTemplate((prev) =>
+			currentSectionData
+				? currentSectionData?.template
+				: prev.template ?? ''
+		);
+
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, []);
+
+	const deleteSection = () => {
+		setSections((prevSections) => {
+			const newSections = prevSections.filter(
+				(prevSection) => prevSection.identifier !== identifier
+			);
+
+			updateSelectedItem(newSections);
+
+			return newSections;
+		});
+	};
+
+	const selectExecutionType = (item) => {
+		if (item.name && item.template && item.executionType) {
+			setSections((prev) => {
+				prev[index] = {
+					...prev[index],
+					...item,
+				};
+
+				updateSelectedItem(prev);
+
+				return prev;
+			});
+		}
+	};
+
+	const updateSelectedItem = (values) => {
+		setSelectedItem((previousItem) => ({
+			...previousItem,
+			data: {
+				...previousItem.data,
+				actions: {
+					description: values.map(({description}) => description),
+					executionType: values.map(
+						({executionType}) => executionType
+					),
+					name: values.map(({name}) => name),
+					priority: values.map(({priority}) => priority),
+					sectionsData: values.map((values) => values),
+					template: values.map(({template}) => template),
+				},
+			},
+		}));
+	};
+
+	return (
+		<SidebarPanel panelTitle={Liferay.Language.get('information')}>
+			<ClayForm.Group>
+				<label htmlFor="name">
+					{Liferay.Language.get('name')}
+
+					<span className="ml-1 mr-1 text-warning">*</span>
+				</label>
+
+				<ClayInput
+					id="name"
+					onBlur={() =>
+						selectExecutionType({
```

Comment:
@alinedoleron I think there's a more efficient way to save each input information. We can think about it together :)

Reply:
@alinedoleron You can ignore the comment above, but I think this function should be renamed as it's not being used only for the executionType. It could be something like "updateActionInfo". What do you think?